### ISSUE TITLE
[Feature Request] Return bool to indicate if image has been flipped.

### ISSUE THREAD

**Issue by varunagrawal** [pytorch/vision#596]
When using the `HorizontalFlip` transform, it can be useful to know via a boolean if the image has been flipped or no. This is in the case of, say, object detection, where the bounding boxes would need to be flipped as well.

We can either define a new function, or add an argument `indicator` that sets the flag whether to return the boolean or not, thus maintaining backwards compatibility.

**Comment by sotte** [comment]
I think you might want to have a look at the functional transforms: https://pytorch.org/docs/stable/torchvision/transforms.html#module-torchvision.transforms.functional

The problem you describe is common for object detection and segmentation. It's not specific to `HorizontalFlip` but applies to all transforms. You want to use the transformation with the *same* parameters to the image and the label.

It's easy to build you transform pipeline with functional transforms. I would look something like this.
```python
import torchvision.transforms.functional as TF
...
if np.random.rand() > .5:
   img, label_img = TF.hflip(img), TF.hflip(label_img)
```
I haven't seen it in combination with `Compose` though.

The [data loader tutorial](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html) might be interesting as well.

**Comment by fmassa** [comment]
I think I'd rather follow the functional interface for that use-case, as pointed out by @sotte.
If one adds a return parameter to `HorizontalFlip` we would also need somewhere to read the returned value to decide if we want to flip or not, which would make the whole API more complex.

I tried out a number of different APIs for taking a number of different input data, but in the end, using the functional API (eventually wrapping it out on classes for your particular need) turned out to be the best and most generic way, and also simplest.

I'm closing this issue, but feel free to comment back if you disagree.
